api/receiver/receive_results.go

```diff
@@ -117,57 +124,71 @@ func HandleResultsUpload(a API, w http.ResponseWriter, r *http.Request) {
 }
 
 func sendResultsToProcessor(
-	a API, uploader string, results int, getFile func(int) (io.ReadCloser, error),
+	a API, uploader string,
+	results int, getResult func(int) (io.ReadCloser, error),
+	screenshots int, getScreenshot func(int) (io.ReadCloser, error),
 	extraParams map[string]string) (*taskqueue.Task, error) {
+
+	if uploader == "" {
+		return nil, fmt.Errorf("empty uploader")
+	}
 	if results == 0 {
 		return nil, fmt.Errorf("nothing uploaded")
 	}
 
 	id := uuid.New()
-	var payloadType string
-	gcs := make([]string, 0, results)
-	if results > 1 {
-		payloadType = "multiple"
-		for i := 0; i < results; i++ {
-			gcsPath := fmt.Sprintf("/%s/%s/%s/%d.json", BufferBucket, uploader, id, i)
-			gcs = append(gcs, gcsPath)
-		}
-	} else {
-		payloadType = "single"
-		gcsPath := fmt.Sprintf("/%s/%s/%s.json", BufferBucket, uploader, id)
-		gcs = append(gcs, gcsPath)
+	resultGCS := make([]string, results)
+	screenshotGCS := make([]string, screenshots)
+	for i := 0; i < results; i++ {
+		resultGCS[i] = fmt.Sprintf("/%s/%s/%s/%d.json", BufferBucket, uploader, id, i)
+	}
+	for i := 0; i < screenshots; i++ {
+		screenshotGCS[i] = fmt.Sprintf("/%s/%s/%s/%d.db", BufferBucket, uploader, id, i)
 	}
 
-	errors := make(chan error, results)
-	var wg sync.WaitGroup
-	wg.Add(results)
-	for i, gcsPath := range gcs {
-		go func(i int, gcsPath string) {
-			defer wg.Done()
-			f, err := getFile(i)
-			if err != nil {
-				errors <- err
-				return
-			}
-			defer f.Close()
-			// TODO: Detect whether the fetched blob is gzipped.
-			if err := a.UploadToGCS(gcsPath, f, true); err != nil {
-				errors <- err
-			}
-		}(i, gcsPath)
+	moveFile := func(wg *sync.WaitGroup, errors chan error, getFile func(int) (io.ReadCloser, error), i int, gcsPath string) {
+		defer wg.Done()
+		f, err := getFile(i)
+		if err != nil {
+			errors <- err
+			return
+		}
+		defer f.Close()
+		// TODO: Detect whether the fetched blob is gzipped.
+		if err := a.UploadToGCS(gcsPath, f, true); err != nil {
+			errors <- err
+		}
 	}
-	wg.Wait()
-	close(errors)
 
-	var errStr string
-	for err := range errors {
-		errStr += strings.TrimSpace(err.Error()) + "\n"
+	var wg1, wg2 sync.WaitGroup
```

Comment:
What's the advantage of a second WaitGroup? Can't we just increase the size of one?